### DISCUSSION TITLE
How to access a dynamic table?

### DISCUSSION THREAD

**Discussion by wingerse** [sqlc-dev/sqlc#2830] [Q&A]
I have a bunch of sharded tables with name of the format: `table_001`, `table_002`, ... `table_999`. 
The correct table is accessed by key % 1000. 

So a query might look as follows:
```
select * from table_%03s where key=?
```
Before executing the query I format it with fmt.Sprintf(query, key).  

Is there a way to do this in sqlc?

**Comment by kyleconroy** [comment]
Currently, there isn't a way to do this. Most database engines do not allow you to use placeholder values in table names.

Can I ask what database engine you're using?

**Reply by wingerse** [reply]
I'm using MySQL. Although it doesn't allow placeholders, formatting the SQL query does the same job for me.  
I was wondering if it's possible to achieve this using a manual extension such as `sqlc.interpolate`?

**Reply by wingerse** [reply]
Can I do a pull request? Lack of support for sharded tables is a blocker to using this in my team..
